native/src/widget/text_input.rs

```diff
@@ -827,7 +880,40 @@ pub fn draw<Renderer>(
     let value = secure_value.as_ref().unwrap_or(value);
 
     let bounds = layout.bounds();
-    let text_bounds = layout.children().next().unwrap().bounds();
+    let text_bounds = {
+        let bounds = layout.children().next().unwrap().bounds();
+        if let Some(icon) = icon {
+            let Icon {
+                font,
+                size,
+                code_point,
+                position,
+            } = icon;
+
+            let padding = f32::from(padding.horizontal());
+            let size = size.unwrap_or_else(|| renderer.default_size());
+            let width = renderer.measure_width(
+                &code_point.to_string(),
+                size,
+                font.clone(),
+            );
+
+            match position {
+                IconPosition::Left => Rectangle {
+                    x: bounds.x + (width + padding),
+                    width: bounds.width - (width + padding),
+                    ..bounds
+                },
+                IconPosition::Right => Rectangle {
+                    x: bounds.x,
+                    width: bounds.width - (width + padding),
+                    ..bounds
+                },
+            }
+        } else {
+            bounds
+        }
+    };
```

Comment:
Could we compute this only once in `layout`? We could add the icon child node after the `text` child, that way event logic won't need to change at all.